### PR TITLE
Fix compiling with XL

### PR DESCRIPTION
**Pull request by masterleinad**
These are the changes required to compile using `XL` and run the test suite.
I am happy to drop the second commit that is related to the tests or do feature detection instead.

### CODE REVIEW

**Comment by masterleinad** [comment]
> Does that mean XL only partially support C++14?

Yes, seemingly. https://www.ibm.com/support/knowledgecenter/en/SSXVZZ_16.1.1/com.ibm.xlcpp1611.lelinux.doc/language_ref/standard_features.html says
> IBM supports selected features of C++14 standard. IBM will continue to develop and implement the features of this standard. The implementation of the language level is based on IBM's interpretation of the standard. Until IBM's implementation of all the C++14 features is complete, including the support of a new C++14 standard library, the implementation might change from release to release. IBM makes no attempt to maintain compatibility, in source, binary, or listings and other compiler interfaces, with earlier releases of IBM's implementation of the new C++14 features.

**Comment by aprokop** [comment]
https://en.cppreference.com/w/cpp/compiler_support#cpp14

File: src/details/ArborX_DetailsBatchedQueries.hpp [hunk 1]
```diff
@@ -91,17 +91,18 @@ struct BatchedQueries
   static auto applyPermutation(Kokkos::View<size_t const *, DeviceType> permute,
                                Predicates const &v)
       -> Kokkos::View<
-          std::decay_t<
+          typename std::decay<
               decltype(Traits::Access<Predicates, Traits::PredicatesTag>::get(
-                  std::declval<Predicates const &>(), std::declval<int>()))> *,
+                  std::declval<Predicates const &>(),
+                  std::declval<int>()))>::type *,
```
Comment (sslattery) [inline src/details/ArborX_DetailsBatchedQueries.hpp]: `clang format`?

Comment (masterleinad) [reply src/details/ArborX_DetailsBatchedQueries.hpp]: https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/ArborX/detail/PR-132/1/pipeline

Comment (sslattery) [reply src/details/ArborX_DetailsBatchedQueries.hpp]: I was more curious why the format changed - not that it was incorrectly formatted

**Comment by dalg24** [comment]
No merging the without my approval.

**Comment by aprokop** [comment]
@dalg24 Please "Request changes" as a review.

**Comment by dalg24** [comment]
> @dalg24 Please "Request changes" as a review.

I am not requesting change.  I am blocking it.

**Comment by masterleinad** [comment]
> @dalg24 Please "Request changes" as a review.

@aprokop Can you just do it yourself please?

**Comment by aprokop** [comment]
We should use [feature macros](https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros) for clarity for situations like this.

**Comment by aprokop** [comment]
Why we care: SIERRA codes are generally required to support the specialized compilers for ASC machines (e.g., Cray and IBM).

**Comment by aprokop** [comment]
Closing as no there has been no activity or interest in the past 8 months. Reopen as needed.

**Pull request closed without merging**